Getting telemetry in parallel fails due to race condition

### Checklist

- [x] I have looked into the [README](https://github.com/openfga/go-sdk/blob/main/README.md) and have not found a suitable solution or answer.
- [x] I have looked into the [documentation](https://openfga.dev/docs) and have not found a suitable solution or answer.
- [x] I have searched the [issues](https://github.com/openfga/go-sdk/issues) and have not found a suitable solution or answer.
- [x] I have upgraded to the [latest version](https://github.com/openfga/go-sdk/releases/latest) of OpenFGA and the issue still persists.
- [x] I have searched the [Slack community](https://openfga.dev/community) and have not found a suitable solution or answer.
- [x] I agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md).

### Description

Getting the telemetry in parallel fails due to a race condition

### Expectation

Getting the telemetry in parallel should not fail

### Reproduction

1. Add the following test to `telemetry/telemetry_test.go`

```go
// Run this test with the "-race" flag.
//
//	go test -race -run ^TestGetParallel$ github.com/openfga/go-sdk/telemetry
func TestGetParallel(t *testing.T) {
	config := &Configuration{}
	factoryParams := TelemetryFactoryParameters{Configuration: config}

	// Error group collects goroutines and receives any error.
	// Context is canceled the first time a goroutine returns a non-nil error.
	eg, ctx := errgroup.WithContext(context.Background())

	// egLimit limits the number of active goroutines
	egLimit := 12
	eg.SetLimit(egLimit)

	// Invokes Get in multiple goroutines. The number of iterations exceeds the
	// error limit in order to test blocking behavior while waiting for earlier
	// goroutines to complete.
	for i := 0; i <= egLimit+2; i++ {
		eg.Go(func() error {
			select {
			case <-ctx.Done():
				return errors.New("Context done early")
			default:
				telemetry := Get(factoryParams)

				if telemetry == nil {
					return errors.New("Expected telemetry to be non-nil")
				}

				if telemetry.Configuration != config {
					return errors.New("Expected configuration to be set correctly")
				}

				return nil
			}
		})
	}

	// Wait() blocks until all goroutines complete.
	// Returns the first non-nil error (if any).
	err := eg.Wait()
	if err != nil {
		t.Fatalf("%v", err)
	}
}
```

2. Run the test with `-race` flag:

```bash
go test -race -run ^TestGetParallel$ github.com/openfga/go-sdk/telemetry
```

3. Observe the error:

```bash
==================
WARNING: DATA RACE
Read at 0x000102553900 by goroutine 8:
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:68 +0x3c
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Previous write at 0x000102553900 by goroutine 19:
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:69 +0x5c
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Goroutine 8 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40

Goroutine 19 (finished) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40
==================
==================
WARNING: DATA RACE
Read at 0x00c000096d50 by goroutine 13:
  runtime.mapaccess1_fast64()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/runtime/map_fast64.go:13 +0x1bc
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:72 +0xbc
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Previous write at 0x00c000096d50 by goroutine 19:
  runtime.mapaccess2_fast64()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/runtime/map_fast64.go:53 +0x1cc
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:74 +0xfc
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Goroutine 13 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40

Goroutine 19 (finished) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40
==================
==================
WARNING: DATA RACE
Read at 0x00c0000be168 by goroutine 8:
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:72 +0xc4
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Previous write at 0x00c0000be168 by goroutine 19:
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:74 +0x108
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Goroutine 8 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40

Goroutine 19 (finished) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40
==================
==================
WARNING: DATA RACE
Read at 0x00c00000e310 by goroutine 13:
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:126 +0xd8
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Previous write at 0x00c00000e310 by goroutine 19:
  github.com/openfga/go-sdk/telemetry.Configure()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:57 +0x1c8
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:73 +0xd4
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Goroutine 13 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40

Goroutine 19 (finished) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40
==================
--- FAIL: TestGetParallel (0.00s)
    testing.go:1465: race detected during execution of test
FAIL
FAIL	github.com/openfga/go-sdk/telemetry	0.743s
FAIL
```

### OpenFGA SDK version

0.6.0 and 0.6.1

### OpenFGA version

any

### SDK Configuration

tests

### Logs

```
==================
WARNING: DATA RACE
Read at 0x000102553900 by goroutine 8:
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:68 +0x3c
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Previous write at 0x000102553900 by goroutine 19:
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:69 +0x5c
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Goroutine 8 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40

Goroutine 19 (finished) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40
==================
==================
WARNING: DATA RACE
Read at 0x00c000096d50 by goroutine 13:
  runtime.mapaccess1_fast64()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/runtime/map_fast64.go:13 +0x1bc
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:72 +0xbc
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Previous write at 0x00c000096d50 by goroutine 19:
  runtime.mapaccess2_fast64()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/runtime/map_fast64.go:53 +0x1cc
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:74 +0xfc
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Goroutine 13 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40

Goroutine 19 (finished) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40
==================
==================
WARNING: DATA RACE
Read at 0x00c0000be168 by goroutine 8:
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:72 +0xc4
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Previous write at 0x00c0000be168 by goroutine 19:
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:74 +0x108
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Goroutine 8 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40

Goroutine 19 (finished) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40
==================
==================
WARNING: DATA RACE
Read at 0x00c00000e310 by goroutine 13:
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:126 +0xd8
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Previous write at 0x00c00000e310 by goroutine 19:
  github.com/openfga/go-sdk/telemetry.Configure()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:57 +0x1c8
  github.com/openfga/go-sdk/telemetry.Get()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry.go:73 +0xd4
  github.com/openfga/go-sdk/telemetry.TestGetParallel.func1()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:120 +0xc4
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x7c

Goroutine 13 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40

Goroutine 19 (finished) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /Users/jim.anderson/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x10c
  github.com/openfga/go-sdk/telemetry.TestGetParallel()
      /Users/jim.anderson/code/fga/sdk-generator/clients/fga-go-sdk/telemetry/telemetry_test.go:115 +0x9c
  testing.tRunner()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/jim.anderson/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1648 +0x40
==================
--- FAIL: TestGetParallel (0.00s)
    testing.go:1465: race detected during execution of test
FAIL
FAIL	github.com/openfga/go-sdk/telemetry	0.743s
FAIL
```

### References

_No response_

--- a/telemetry/telemetry.go
+++ b/telemetry/telemetry.go
@@ -2,6 +2,7 @@ package telemetry
 
 import (
 	"context"
+	"sync"
 
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/metric"
@@ -42,6 +43,9 @@ type QueryDurationMetricParameters struct {
 type TelemetryContextKey struct{}
 
 var (
+	telemetryInstancesLock sync.Mutex
+	// Warning: do not use directly, it may cause data race.
+	// Deprecated: this map will be renamed to telemetryInstances.
 	TelemetryInstances map[*Configuration]*Telemetry
 	TelemetryContext   TelemetryContextKey
 )
@@ -64,6 +68,9 @@ func Get(factory TelemetryFactoryParameters) *Telemetry {
 	if configuration == nil {
 		configuration = DefaultTelemetryConfiguration()
 	}
+
+	telemetryInstancesLock.Lock()
+	defer telemetryInstancesLock.Unlock()
 
 	if TelemetryInstances == nil {
 		TelemetryInstances = make(map[*Configuration]*Telemetry)

--- a/telemetry/telemetry_test.go
+++ b/telemetry/telemetry_test.go
@@ -3,6 +3,7 @@ package telemetry
 import (
 	"context"
 	"net/http"
+	"sync"
 	"testing"
 	"time"
 
@@ -224,4 +225,25 @@ func TestBuildTelemetryAttributesMethod(t *testing.T) {
 	if requestDuration != 100 {
 		t.Fatalf("Expected requestDuration to be 100, but got %v", requestDuration)
 	}
+}
+
+// Run this test with the "-race" flag.
+//
+//	go test -race -run ^TestGetRace$ github.com/openfga/go-sdk/telemetry
+func TestGetRace(t *testing.T) {
+	t.Parallel()
+
+	var wg sync.WaitGroup
+
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+
+		go func() {
+			defer wg.Done()
+
+			Get(TelemetryFactoryParameters{})
+		}()
+	}
+
+	wg.Wait()
 }

